lib/PackageEnvironment.js

```diff
@@ -184,14 +184,19 @@ function builtInsPerPackage(
   }
   return {
     [`${prefix}__name`]: builtIn(
-      packageJson.name
+      name
     ),
     [`${prefix}__version`]: builtIn(
-      packageJson.version || null
+      version || null
     ),
-    [`${prefix}__root`]: builtIn(
+    [`${prefix}__source_root`]: builtIn(
       relativeToSandbox(sandbox.packageInfo.rootDirectory, rootDirectory)
     ),
+    [`${prefix}__root`]: builtIn(
+      esy.buildOutOfSource
+        ? relativeToSandbox(sandbox.packageInfo.rootDirectory, rootDirectory)
+        : targetPath(sandbox, packageInfo, '_build')
```

Comment:
If package can't build out of source then we copy sources over to `$cur__target_dir` and set `$cur__root` to be equal to `$cur__target_dir`.